### DISCUSSION TITLE
preprocessor markdown not working on build

### DISCUSSION THREAD

**Discussion by natapg** [gatsbyjs/gatsby#34013] [Help]
Hi! I've migrated to gatsby v4 and now my preprocessor is not working on build but it works on development, so I think is related to the new changes on building process, but I couldn't find the cause root.

My preprocessor is this (gatsby-node)

```
const remark = require('remark')
const remarkHTML = require('remark-html')

function preprocessMarkdown(obj) {
  if (obj && typeof obj === 'object') {
    Object.keys(obj).forEach((key) => {
      if (key.startsWith('markdown__')) {
        obj[key] = remark().use(remarkHTML).processSync(obj[key]).toString()
      } else {
        // We need to preprocess subobjects as well.
        preprocessMarkdown(obj[key])
      }
    })
  }
}

exports.onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions

  if (node.internal.type === `MarkdownRemark`) {
    const value = createNodePath({ node, getNode })
    createNodeField({
      name: `slug`,
      node,
      value,
    })
  }

  // Replace the value of all markdown fields by the parsed HTML.
  const frontmatter = node.frontmatter
  preprocessMarkdown(frontmatter)
}
```
If I add a console.log I can check that during the building process the markdown is converted to html, but then the query result is markdown instead of the html code,do you know where the problem can be?

**Comment by raae** [comment]
You are not allowed to change a node. You can only add fields. 
See https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#dont-mutate-nodes-outside-of-expected-apis
